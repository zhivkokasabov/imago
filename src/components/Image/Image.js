import React from 'react';

import classNames from 'classnames';

import styles from './Image.module.scss';

export default function Image({
  omitDomain,
  ...imgElementAttributes
}) {
  return <img
    {...imgElementAttributes}
    src={omitDomain ? imgElementAttributes.src : `${process.env.CDN}${imgElementAttributes.src}`}
    className={classNames(imgElementAttributes.className, styles.image)}
  />;
}
