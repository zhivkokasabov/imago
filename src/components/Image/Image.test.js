import React from 'react';
import { render } from '@testing-library/react';

import Image from './Image';

describe('Image component', () => {
  it('should render image', () => {
    const { asFragment } = render(<Image src="logo.svg" width={100} height={100} loading="lazy" srcSet="logo.svg" sizes="sizes" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
