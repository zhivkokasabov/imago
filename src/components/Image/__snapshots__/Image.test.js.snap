// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render image 1`] = `
<DocumentFragment>
  <img
    height="100"
    loading="lazy"
    sizes="sizes"
    src="logo.svg"
    srcset="logo.svg"
    width="100"
  />
</DocumentFragment>
`;
